blog/2023-10-25-implementing-sign-in-with-google.md

```diff
@@ -0,0 +1,519 @@
+and sign in with the Google account that should manage the app and credential tokens.
+
+* Click "Create Project" and give it a name. After creation the new project should be selected.
+* Click "Configure Consent Screen", Choose "External", then Create.
+  * Enter App Name and User Support Email -- these will be shown to users when logging in
+  * Scroll all the way down to "Developer contact information" and add your email address, click "Save and Continue"
+  * Click "Add or Remove Scopes"
+    * Select "Email", "Profile", and "OpenID Connect"
+    * Click "Update", then "Save and Continue"
+  * Add any test users that should be able to log in during development.
+* From the "Credentials" page, click "+ Create Credentials", then "OAuth client ID"
+  * Select Application Type: "Web Application"
+  * Add Authorized Javascript Origins: `http://localhost`, `http://localhost:3000`, `https://example.com` (prod domain must be HTTPS)
+  * Click "Save"
+* Copy the OAuth "Client ID" and save it for later. Mine looks like `309209880368-4uqd9e44h7t4alhhdqn48pvvr63cc5j5.apps.googleusercontent.com`
+
+```python eval
+rx.center(
+    rx.video(url="https://github.com/reflex-dev/reflex-examples/assets/1524005/af2499a6-0bda-4d60-b52b-4f51b7322fd5"),
+    rx.box(height="3em"),
+    width="100%",
+    padding_y="2em"
+)
+```
+
+## Wrap @react-oauth/google
+
+The [`@react-oauth/google`](https://github.com/MomenSherif/react-oauth)
+package provides a React component that handles all of the interaction with
+Google's OAuth API. It has rich functionality and many options, but for the
+purposes of this post, we will only wrap the props needed to get basic login
+working.
+
+### GoogleOAuthProvider
+
+The `GoogleOAuthProvider` component is responsible for downloading Google's SDK
+and supplying the Client ID.
+
+Create a new file in your app directory, `react_oauth_google.py`.
+
+```python
+import reflex as rx
+
+class GoogleOAuthProvider(rx.Component):
+    library = "@react-oauth/google"
+    tag = "GoogleOAuthProvider"
+
+    client_id: rx.Var[str]
+```
+
+```python eval
+docalert(
+    "Why client_id and not clientID",
+    "When Reflex compiles the component to Javascript, `snake_case` property names "
+    "are automatically formatted as `camelCase`, although the names may be defined "
+    "in `camelCase` as well",
+)
+```
+### GoogleLogin
+
+The `GoogleLogin` component renders the familiar "Sign in with Google" button.
+
+Since we will use the default configuration, no props are needed, however the
+event trigger does need to be defined so our Reflex app is able to get the token
+after logging in.
+
+Define the following wrapper in the same `react_oauth_google.py`.
+
+```python
+class GoogleLogin(rx.Component):
+    library = "@react-oauth/google"
+    tag = "GoogleLogin"
+
+    def get_event_triggers(self):
+        return \{"on_success": lambda data: [data]}
```

Comment:
Not 100% clear how the `on_success` trigger here relates to the `on_success` State function defined on line 145

Reply:
There is a line in the frontend code that associates them

```python
GoogleLogin.create(on_success=State.on_success)
```

Reply:
Ah yes I added this comment before I got down to this line in the code

Reply:
do you think it's okay, or should i name the state event handler something else, like `handle_token`?